src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,173 @@
+            String taskDetails = taskString.substring(descriptionIndex);
+
+            if (taskString.charAt(4) == '1') {
+                isDone = true;
+            }
+
+            String eventOrDeadlineDescription = "";
+            if (taskType == 'E' || taskType == 'D') {
+                eventOrDeadlineDescription = taskString.substring(8, taskString.lastIndexOf("|") - 1);
+            }
+
+            switch (taskType) {
+            case 'T':
+                taskList.add(new ToDo(taskDetails, isDone));
+                break;
+            case 'E':
+                taskList.add(new Event(eventOrDeadlineDescription, isDone, formatTaskDateTime(taskDetails)));
+                break;
+            case 'D':
+                taskList.add(new Deadline(eventOrDeadlineDescription, isDone, formatTaskDateTime(taskDetails)));
+                break;
+            default:
+                System.out.println("Unable to determine type of task");
+                break;
+            }
+        }
+
+        return taskList;
+    }
+
+    private String formatTaskDateTime(String dateTime) {
+        assert dateTime.length() != 0 : "Date and time cannot be empty";
+        String[] dateTimes = dateTime.split(",");
+        String date = dateTimes[0]; // MMM DD YYYY
+        String time = dateTimes[1]; // HH:MM:SS
+
+        DateTimeFormatter format = DateTimeFormatter.ofPattern("MMM d yyyy");
+        LocalDate newDate = LocalDate.parse(date, format);
+        int monthInt = newDate.getMonthValue();
+        String year = String.valueOf(newDate.getYear());
+        int dayInt = newDate.getDayOfMonth();
+
+        String month = monthInt >= 10
+                ? String.valueOf(monthInt)
+                : "0".concat(String.valueOf(monthInt));
+
+        String day = dayInt >= 10
+                ? String.valueOf(dayInt)
+                : "0".concat(String.valueOf(dayInt));
+
+        String hourAndMinutes = time.substring(0, time.lastIndexOf(":"));
+
+        // Return YYYY/MM/DD HH:MM
+        String ans = year + "/" + month + "/" + day + hourAndMinutes;
+        return ans;
+    }
+
+    /**
+     * Saves all the Tasks from the TaskList into the user's local storage.
+     */
+    public void writeToStorage() {
+        try {
+            assert pathToStorage != null : "Path to storage cannot be empty";
+            File storageFile = new File(String.valueOf(pathToStorage));
+            FileWriter fw = new FileWriter(storageFile);
+            for (String task : this.allTasks) {
+                fw.write(task + '\n');
+            }
+            fw.close();
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    /**
+     * Adds a new Task to the local storage.
+     *
+     * @param task the Task to be added.
+     */
+    public void createTask(Task task) {
+        assert task != null : "Task cannot be null";
+        this.allTasks.add(task.toString());
```

Comment:
If I recall correctly, the code standard suggests not to use 'this' unless there is ambiguity in the word (like in the constructor) so maybe can remove 'this' over here?